src/general/ShuffleBag.cs

```diff
@@ -0,0 +1,137 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+public class ShuffleBag<T> : IEnumerable<T>
```

Comment:
As this looks like a class designed for more general use, this should be documented better